src/matchbox/matchbox.py

```diff
@@ -15,106 +15,35 @@
-    +-----------+------------------+
-    | Any new   | Ob2, Ob5         |
-    +-----------+------------------+
-
-
-    Example for multi-trait characteristic:
-
-    +--------+-------+-------------------------+
-    | Entity | Match | Characteristic's traits |
-    +========+=======+=========================+
-    | Ob1    | False | 1, 2                    |
-    +--------+-------+-------------------------+
-    | Ob2    | True  | 3, 4, 7                 |
-    +--------+-------+-------------------------+
-    | Ob3    | False | 5, 6                    |
-    +--------+-------+-------------------------+
-    | Ob4    | False |                         |
-    +--------+-------+-------------------------+
-    | Ob5    | True  | 1, 7                    |
-    +--------+-------+-------------------------+
-
-    Should result in this index:
-
-    +-----------+------------------+
-    | Attribute | Matched Entities |
-    +===========+==================+
-    | 1         | Ob1, Ob2         |
-    +-----------+------------------+
-    | 2         | Ob1, Ob2, Ob5    |
-    +-----------+------------------+
-    | 3         | Ob5              |
-    +-----------+------------------+
-    | 4         | Ob5              |
-    +-----------+------------------+
-    | 5         | Ob2, Ob3, Ob5    |
-    +-----------+------------------+
-    | 6         | Ob2, Ob3, Ob5    |
-    +-----------+------------------+
-    | 7         |                  |
-    +-----------+------------------+
-    | Any other | Ob2, Ob5         |
-    +-----------+------------------+
-
-    """
-
-    def __init__(self, characteristic):
+class MatchBox(MatchIndex):
+    """MatchBox is a MatchIndex that can index objects by their fields."""
+
+    def __init__(self, characteristic, *args, **kwargs):
         """
-        Initialise box and set attribute this box will be indexing objects on.
+        Initialise the box and set the attribute this box will be indexing objects on.
 
-        Indexed entities are expected to have attribute of the same name as
-        `characteristic` that contains traits by which object is classified.
-        Optionally entity should have second attribute that will state that
-        the object is classified by the trait **OR** that he object is not
-        classified by the trait called **{characteristic}_match**.
+        Indexed entities are expected to have an attribute of the same name as
+        `characteristic` that contains a trait by which entity is classified.
+        Optionally entities may have a second attribute stating whether
+        the object should be classified to match this trait or the object should be
+        classified to mismatch the trait. This atribute, for each characteristic, is called
+        **{characteristic}_match**.
 
         .. note::
 
+            Each indexed entity has to have at least one characteristic.
+
             You can say that entity's characteristic is, or is not described by
```

Comment:
Okay, it would be `described by some traits` (the is part) and `described by all except given traits` (is not part). Which sounds better than the current wording.

Anyway that's the reason I'd like to have two pull requests actually, one with docstrings being rewritten and second with extracting indexing logic ;)